internal/cloudrunci/cloudrunjobs.go

```diff
@@ -0,0 +1,324 @@
+		return errors.New("Project ID missing")
+	}
+	if err := j.Env.Validate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// version returns the revision that the service will be deployed to.
+// This identifier is also used to locate relevant log messages.
+func (j *RunJob) version() string {
+	return j.Name + "-" + runID
+}
+
+// Create deploys the job to Cloud Run Jobs, but does not start it.
+// If an image has not been specified or previously built, it will call Build.
+func (j *RunJob) Create() error {
+	// Don't deploy unless we're certain everything is ready for deployment
+	// (i.e. admin client is authenticated and authorized)
+	if err := j.validate(); err != nil {
+		return err
+	}
+
+	if j.Image == "" && !j.built {
+		if err := j.Build(); err != nil {
+			return err
+		}
+	}
+
+	if _, err := gcloud(fmt.Sprintf("%s: Creating Cloud Run Job", j.version()), j.createCmd()); err != nil {
+		return fmt.Errorf("gcloud: %s: %q", j.version(), err)
+	}
+
+	j.created = true
+	return nil
+}
+
+// Build builds a container image if one has not already been built.
+// If service.Image is specified and this is directly called, it
+// could overwrite an existing container image.
+func (j *RunJob) Build() error {
+	if err := j.validate(); err != nil {
+		return err
+	}
+	if j.built {
+		return fmt.Errorf("container image already built")
+	}
+	if j.Image == "" {
+		j.Image = fmt.Sprintf("gcr.io/%s/%s:%s", j.ProjectID, j.Name, runID)
+	}
+
+	if _, err := gcloud(fmt.Sprintf("%s: Building image %s", j.version(), j.Image), j.buildCmd()); err != nil {
+		return fmt.Errorf("gcloud: %s: %q", j.Image, err)
+	}
+	j.built = true
+
+	return nil
+}
+
+// Run starts the Job in Cloud Run Jobs.
+// This method will call Build and Create if necessary.
+func (j *RunJob) Run() error {
+	if err := j.validate(); err != nil {
+		return err
+	}
+	// Create() checks that the image was built
+	if !j.created {
+		if err := j.Create(); err != nil {
+			return err
+		}
+	}
+	if _, err := gcloud(fmt.Sprintf("%s: Running cloud run job", j.version()), j.runCmd()); err != nil {
+		return fmt.Errorf("gcloud: %v: %q", j.version(), err)
+	}
+	return nil
+}
+
+// Clean deletes the created Cloud Run service.
+func (j *RunJob) Clean() error {
+	// NOTE: don't check whether j.created is set.
+	// We may want to attempt to clean up if creation failed (i.e. to clean up the image).
+
+	if err := j.validate(); err != nil {
+		return err
+	}
+
+	if _, err := gcloud(fmt.Sprintf("%s: Deleting cloud run job", j.version()), j.deleteRunJobCmd()); err != nil {
+		return fmt.Errorf("gcloud: %v: %q", j.version(), err)
+	}
+	j.created = false
+
+	// If built is false, no image was created or is not managed by cloudrun-ci.
+	if j.built {
+		_, err := gcloud(fmt.Sprintf("%s: Deleting Image %s", j.version(), j.Image), j.deleteImageCmd())
+		if err != nil {
+			return fmt.Errorf("gcloud: %v: %q", j.version(), err)
+		}
+		j.built = false
+	}
+
+	return nil
+}
+
+func (j *RunJob) createCmd() *exec.Cmd {
```

Comment:
In the future, we may need to support flags, such as tasks, parallelism and retries